sdk/include/cheri.h

```diff
@@ -0,0 +1,21 @@
+// Copyright CHERIoT Contributors.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+#include <stddef.h>
+#include <stdint.h>
+#include <compartment-macros.h>
+
+/**
+ * C API for `check_pointer`.
+ *
+ * This function behaves identically to `CHERI::check_pointer`, but is exposed
+ * without templating and C++ types.
+ *
+ * To reduce code size, this function is provided as part of the compartment
+ * helper library.
```

Comment:
Please can you document this without reference to the C++ version (or with just a note for C++ people that they should consider the C++ version instead)?  C programmers shouldn't have to read C++ code to understand what C APIs do.